src/tools/qgepnetwork.py

```diff
@@ -163,34 +167,37 @@ def refresh(self):
         Refreshes the network graph. It will force a refresh of the materialized views in the database and then reload
         and recreate the graph.
         """
-        uri = QgsDataSourceUri(self.nodeLayer.dataProvider().dataSourceUri())
+        with OverrideCursor(Qt.WaitCursor):
+            transaction = self.nodeLayer.dataProvider().transaction()
+            temporary_edit_session = False
+            if not transaction:
+                self.nodeLayer.startEditing()
+                temporary_edit_session = True
+                transaction = self.nodeLayer.dataProvider().transaction()
 
-        db = QSqlDatabase.addDatabase("QPSQL")  # Name of the driver -- doesn't change
+                if not transaction:
+                    self.message_emitted.emit(self.tr("Warning"), self.tr("Could not initialize transaction"), Qgis.Critical)
```

Comment:
should be error instead of Warning? at least text does not correspond to error level